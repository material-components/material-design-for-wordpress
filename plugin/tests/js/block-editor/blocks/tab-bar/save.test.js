/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Save from '../../../../../assets/src/block-editor/blocks/tab-bar/save';

/**
 * Render the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Save { ...props } /> );
};

const tabProps = {
	attributes: {
		tabs: [
			{
				label: 'Tab 1 Title',
				content: [
					{
						clientId: 'c0cf3523-d407-4389-939d-b1d1007b316f',
						name: 'core/paragraph',
						isValid: true,
						attributes: {
							content: 'Tab 1 content',
							dropCap: false,
						},
						innerBlocks: [],
					},
				],
			},
			{
				label: 'Tab 2 Title',
				content: [
					{
						clientId: '00d62a82-e010-49cb-982a-bc04767df7ea',
						name: 'core/paragraph',
						isValid: true,
						attributes: {
							content: 'Tab 2 content',
							dropCap: false,
						},
						innerBlocks: [],
					},
				],
			},
		],
		iconPosition: 'none',
	},
	className: 'wp-block-material-tab-bar',
};

describe( 'blocks: material/tab-bar: Save', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( tabProps );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
