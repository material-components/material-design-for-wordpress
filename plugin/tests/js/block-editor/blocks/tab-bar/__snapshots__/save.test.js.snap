// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/tab-bar: Save matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-tab-bar-container"
      >
        <div
          class="mdc-tab-bar"
          role="tablist"
        >
          <div
            class="mdc-tab-scroller"
          >
            <div
              class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            >
              <div
                class="mdc-tab-scroller__scroll-content"
              >
                <div
                  class="mdc-tab tab mdc-tab--active"
                  role="tab"
                  tabindex="0"
                >
                  <span
                    class="mdc-tab__content"
                  >
                    <span
                      class="mdc-tab__text-label tab__label-field"
                    >
                      <span
                        role="tab"
                        tabindex="0"
                      >
                        Tab 1 Title
                      </span>
                    </span>
                  </span>
                  <span
                    class="mdc-tab-indicator mdc-tab-indicator--active"
                  >
                    <span
                      class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    />
                  </span>
                  <span
                    class="mdc-tab__ripple"
                  />
                </div>
                <div
                  class="mdc-tab tab"
                  role="tab"
                  tabindex="0"
                >
                  <span
                    class="mdc-tab__content"
                  >
                    <span
                      class="mdc-tab__text-label tab__label-field"
                    >
                      <span
                        role="tab"
                        tabindex="0"
                      >
                        Tab 2 Title
                      </span>
                    </span>
                  </span>
                  <span
                    class="mdc-tab-indicator"
                  >
                    <span
                      class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    />
                  </span>
                  <span
                    class="mdc-tab__ripple"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="mdc-tab-content mdc-typography--body1 mdc-tab-content--active"
          />
          <div
            class="mdc-tab-content mdc-typography--body1"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-tab-bar-container"
    >
      <div
        class="mdc-tab-bar"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <div
                class="mdc-tab tab mdc-tab--active"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  <span
                    class="mdc-tab__text-label tab__label-field"
                  >
                    <span
                      role="tab"
                      tabindex="0"
                    >
                      Tab 1 Title
                    </span>
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <span
                  class="mdc-tab__ripple"
                />
              </div>
              <div
                class="mdc-tab tab"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  <span
                    class="mdc-tab__text-label tab__label-field"
                  >
                    <span
                      role="tab"
                      tabindex="0"
                    >
                      Tab 2 Title
                    </span>
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <span
                  class="mdc-tab__ripple"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mdc-tab-content mdc-typography--body1 mdc-tab-content--active"
        />
        <div
          class="mdc-tab-content mdc-typography--body1"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
